### PR TITLE
feat: new welcome screen

### LINKED ISSUE #233: Round of UX updates
Here's some UX feedback on the current version of the app:

- [ ] Faucet button should be visible on mobile, now it's hidden
- [ ] Faucet pop-up does not have a loading state for the captcha, which make it confusing when you're waiting on it to load. 
- [ ] If you create your wallet and do not have ETH, there is no logical action to take other than using the fauceting. So we should automatically load the faucet and force the user to use it. We can change the copy for first-time use to "You currently don't have any ETH. To start using the dapp, click below to faucet ETH into your wallet."

### PR DESCRIPTION
**Pull request by pedronauck**
✨ Fix #233

## Tasks

- [x] New revamp of the auto faucet (without tour anymore, but using steps instead, layout bellow)
- [x] Add new animations
- [x] Implement responsive queries on the new design
- [x] Implement UI/UX changes related to the issue #233 

## New Design Spec

https://user-images.githubusercontent.com/2029172/171523030-bc6c9a94-3279-4e57-853d-d3c290caadb6.mov

### CODE REVIEW

**Comment by luizstacio** [comment]
@pedronauck update the `.env.production` with the current contract ids, this would enable better testing on the preview env.

File: packages/app/src/hooks/useWelcomeSteps.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  path: string | null;
+};
+
+type MachineContext = {
+  current: Step;
+};
+
+type MachineEvents = { type: "NEXT" } | { type: "SET_CURRENT"; value: number };
+
+const welcomeStepsMachine = createMachine<MachineContext>({
```
Comment (luizstacio) [inline packages/app/src/hooks/useWelcomeSteps.tsx:53]: ❤️❤️

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/app/.env.production [hunk 2]
```diff
@@ -5,5 +5,5 @@
 VITE_FUEL_PROVIDER_URL=https://node.swayswap.io/graphql
 VITE_FUEL_FAUCET_URL=https://faucet-fuel-core.swayswap.io/dispense
 VITE_FAUCET_RECAPTCHA_KEY=6Ld3cEwfAAAAAMd4QTs7aO85LyKGdgj0bFsdBfre
-VITE_CONTRACT_ID=0x32aa46fbca80077804d74dcb2fbbe737b3d41a2ea3678b712cdd20525eae9911
+VITE_CONTRACT_ID=0x453ca9d17e3faf8f5b32f72b6bb54f0ec7e3c6b538f21ebb4999aa0fdd350aed
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/.env.production:8]: ✅

File: packages/app/src/components/FaucetWidget.tsx [hunk 3]
```diff
@@ -1,107 +1,34 @@
...
-              variant="primary"
-              isFull
-              className="mt-5"
-              isDisabled={!faucetCaptcha && !!RECAPTCHA_KEY}
-              isLoading={faucet.isLoading}
-              onPress={() => faucet.handleFaucet(faucetCaptcha)}
-            >
-              Give me ETH
-            </Button>
+            <div>Click the button below to mint 0.5 ETH to your wallet.</div>
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/src/components/FaucetWidget.tsx]: ✅

File: packages/app/src/hooks/useWelcomeSteps.tsx (hunk 1 above)
Comment (pedronauck) [reply packages/app/src/hooks/useWelcomeSteps.tsx:53]: xstate is pure love 😍

File: packages/app/src/components/FaucetApp.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+export function FaucetApp({ onSuccess }: FaucetAppProps) {
+  const [faucetCaptcha, setFaucetCaptcha] = useState<string | null>(null);
+  const faucet = useFaucet({ onSuccess });
+
+  return (
+    <>
+      {RECAPTCHA_KEY && (
+        <div className="faucetCaptcha">
```
[the remark this answers is not in the document]
Comment (luizstacio) [reply packages/app/src/components/FaucetApp.tsx:21]: I think today it at least shows an error when it happens, that is better they get stuck is not so uncommon to happen here. I would harder put the loading using hard code time... at leat user knows what happened

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/app/src/components/FaucetApp.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/src/components/FaucetApp.tsx:21]: Cool, i'll do that 👊🏻

**Review by luizstacio — APPROVED** [review]

**Pull request merged**